docs/enhancements/add-kernel-parameters.md

```diff
@@ -0,0 +1,81 @@
+---
+title: add-discovery-kernel-arguments
+authors:
+- "@oamizur"
+  creation-date: 2022-09-20
+  last-updated: 2022-09-20
+---
+
+# Add discovery kernel arguments
+
+## Summary
+
+Some environments require additional kernel arguments to boot host machines.  
+We would like to support additional kernel arguments while booting the discovery image.
+
+## Motivation
+
+Previously, adding kernel arguments was required by several customers in order to adapt to various environments.
+For example **rd.net.timeout.carrier** may be required in order for either static networking to be applied or 
+to successfully get a dhcp address before pulling the rootfs.
+### Goals
+Add capability to set additional kernel arguments to discovery image.
+
+Support the following scenarios:
+- Full ISO image
+- Minimal ISO image
+- iPXE
+- Cloud/operator (when considering authentication from image service to assisted-service)
+
+### Non-Goals
+Manipulation of existing kernel parameters (update/delete)
+
+## Proposal
+
+The discovery kernel arguments will be stored in the infra-env database record in a designated JSON encoded field.
+During infra-env registration and update the full set of the kernel arguments can be set.
+
+Setting the arguments can either be done by either REST API or Kube API (Part of InfraEnv CR)
+### User Stories
+
+#### Story 1
+
+When having slow NIC, increase the timeout to have the IP address configured.
+
+### Implementation Details/Notes/Constraints
+
+#### iPXE implementation
+
+iPXE implementation should be quite straight forward.  The arguments will be concatenated at the end of the 
+**kernel** command which is part of the iPXE script.
+
+#### Minimal and full ISO
+
+The embedding of the kernel arguments into the ISO is an issue that needs to be investigated.
```

Comment:
Where does it go in the ISO?  Currently, we know how to embed ignition.

Reply:
Not sure yet

Reply:
There is an embed area reserved for kernel arguments, but it's within a file.
For ignition we currently find the file and use that as the offset to start embedding.

Reply:
When you say "it's within a file", do you mean that it's part of a file as opposed to an entire file?  If so, is there an easy way to determine the offset within that file?

Reply:
> When you say "it's within a file", do you mean that it's part of a file as opposed to an entire file?

Yes

> If so, is there an easy way to determine the offset within that file?

I'm not sure. I can think of some things we might do, but it would be good to investigate what coreos is doing to sort out the "right" way.

Reply:
Added description how it is done